lib/babe/build.go

```diff
@@ -222,22 +222,38 @@ func (b *BlockBuilder) buildBlockExtrinsics(slot Slot, rt runtime.Instance) []*t
 	return included
 }
 
-func buildBlockInherents(slot Slot, rt runtime.Instance) ([][]byte, error) {
+func buildBlockInherents(slot Slot, rt runtime.Instance, parent *types.Header) ([][]byte, error) {
 	// Setup inherents: add timstap0
 	idata := types.NewInherentsData()
 	timestamp := uint64(time.Now().UnixMilli())
-	err := idata.SetInt64Inherent(types.Timstap0, timestamp)
+	err := idata.SetInherent(types.Timstap0, timestamp)
 	if err != nil {
 		return nil, err
 	}
 
 	// add babeslot
-	err = idata.SetInt64Inherent(types.Babeslot, slot.number)
+	err = idata.SetInherent(types.Babeslot, slot.number)
 	if err != nil {
 		return nil, err
 	}
 
-	ienc, err := idata.Encode()
+	parachainInherent := ParachainInherentData{
+		ParentHeader: *parent,
+	}
+
+	// add parachn0 and newheads
+	// for now we can use "empty" values, as we require parachain-specific
+	// logic to actually provide the data.
+
+	if err = idata.SetInherent(types.Parachn0, parachainInherent); err != nil {
+		return nil, fmt.Errorf("setting struct inherent %q: %w", types.Parachn0, err)
+	}
+
+	if err = idata.SetInherent(types.Newheads, []byte{0}); err != nil {
+		return nil, fmt.Errorf("setting struct inherent %q: %w", types.Newheads, err)
```

Comment:
nit since the method name changed to `SetInherent`, maybe this is better:

```suggestion
		return nil, fmt.Errorf("setting inherent %q: %w", types.Newheads, err)
```